app/controllers/authorities_controller.rb

```diff
@@ -20,13 +20,20 @@ def get_results(subject, search)
     lcsh_base_uri = "http://id.loc.gov/authorities/subjects/suggest/?q="
     # can increase limit to max 20 with '&rows=20'
     fast_geo_uri = "http://fast.oclc.org/searchfast/fastsuggest?fl=suggest51&wt=json&query="
+    
     if subject == 'lcsh'
       result = JSON.parse(HTTParty.get(
         lcsh_base_uri + "#{search.gsub(/\s/,'*')}*"
-      ))[1]
+      ))
+      result = result[1] ? result[1] : ["Error Searching"]
     elsif subject == 'subject_geographic'
-      result = HTTParty.get(fast_geo_uri + search)["response"]["docs"]
-      result.map!{ |x| x["suggest51"] }
+      result = HTTParty.get(fast_geo_uri + search)
+      if result["response"]["docs"]
```

Comment:
you can just do `if result = result["response"]["docs"]`

Reply:
ah, i didn't think of this. should i refactor?

Reply:
it's up to you